diff --git a/dlls/winex11.drv/event.c b/dlls/winex11.drv/event.c
index 541eee1..d0a352f 100644
--- a/dlls/winex11.drv/event.c
+++ b/dlls/winex11.drv/event.c
@@ -978,8 +978,14 @@ static void focus_out( Display *display , HWND hwnd )
            case we don't have to change the foreground window to 0 */
         if (hwnd == NtUserGetForegroundWindow())
         {
-            TRACE( "lost focus, setting fg to desktop\n" );
+#ifdef __ANDROID__
+            TRACE( "lost focus on Android, preserving foreground window
+" );
+#else
+            TRACE( "lost focus, setting fg to desktop
+" );
             NtUserSetForegroundWindow( NtUserGetDesktopWindow() );
+#endif
         }
     }
  }
@@ -1119,9 +1125,13 @@ static BOOL X11DRV_MapNotify( HWND hwnd, XEvent *event )
 
     if (!data->managed && !data->embedded && data->desired_state.wm_state != WithdrawnState)
     {
+#ifdef __ANDROID__
+        set_input_focus( data );
+#else
         HWND hwndFocus = get_focus();
         if (hwndFocus && NtUserIsChild( hwnd, hwndFocus ))
             set_input_focus( data );
+#endif
     }
     release_win_data( data );
     return TRUE;
