diff --git a/dlls/winex11.drv/event.c b/dlls/winex11.drv/event.c
index 7e736df..d618a22 100644
--- a/dlls/winex11.drv/event.c
+++ b/dlls/winex11.drv/event.c
@@ -972,8 +972,12 @@ static void focus_out( Display *display , HWND hwnd )
            case we don't have to change the foreground window to 0 */
         if (hwnd == NtUserGetForegroundWindow())
         {
+#ifdef __ANDROID__
+            TRACE( "lost focus on Android, preserving foreground window\n" );
+#else
             TRACE( "lost focus, setting fg to desktop\n" );
             NtUserSetForegroundWindow( NtUserGetDesktopWindow() );
+#endif
         }
     }
  }
@@ -1113,9 +1117,13 @@ static BOOL X11DRV_MapNotify( HWND hwnd, XEvent *event )
 
     if (!data->managed && !data->embedded && data->desired_state.wm_state != WithdrawnState)
     {
+#ifdef __ANDROID__
+        set_input_focus( data );
+#else
         HWND hwndFocus = get_focus();
         if (hwndFocus && NtUserIsChild( hwnd, hwndFocus ))
             set_input_focus( data );
+#endif
     }
     release_win_data( data );
     return TRUE;
@@ -1828,3 +1836,4 @@ static BOOL X11DRV_ClientMessage( HWND hwnd, XEvent *xev )
     TRACE( "no handler found for %ld\n", event->message_type );
     return FALSE;
 }
+

