diff --git a/dlls/shell32/shlfileop.c b/dlls/shell32/shlfileop.c
index f24c7b7..01b2c1e 100644
--- a/dlls/shell32/shlfileop.c
+++ b/dlls/shell32/shlfileop.c
@@ -949,6 +949,9 @@ static void file_entry_init(FILE_ENTRY *file_entry,
     size_t file_name_len = wcslen(file_name) + 1;
     const WCHAR *ptr;
 
+    file_entry->szDirectory = NULL;
+    file_entry->szFilename = NULL;
+
     file_entry->szFullPath = malloc(file_name_len * sizeof(WCHAR));
     wcscpy(file_entry->szFullPath, file_name);
 
